contracts/oracles/sources/oracle.move

```diff
@@ -0,0 +1,38 @@
+module oracles::oracle {
+
+    use sui::object::{Self, UID};
+    use sui::tx_context::{Self,TxContext};
+    use sui::transfer;
+
+    use SupraOracle::SupraSValueFeed::{Self, OracleHolder};
+
+    struct Storage has key, store {
+        id: sui::object::UID,
+        supra_price: u256,
+    }
+
+    fun init(ctx:&mut TxContext) {
+        transfer::share_object(
+            Storage {
+                id:object::new(ctx),
+                supra_price:0,
+            }
+        );
+    }
+
+    public fun set_supra_price(oracle_holder:&OracleHolder, storage: &mut Storage) {
+        let (sui_usd_price,_,_,_) = SupraSValueFeed::get_price(oracle_holder, 90);
+        storage.supra_price = (sui_usd_price as u256) / 1000000000;   
```

Comment:
Move `1000000000` to a constant just like pair code.

Something like: `SUI_FRACTION: u256 = 100_000_000`